yapapi/config.py

```diff
@@ -0,0 +1,45 @@
+from dataclasses import dataclass, field
+from functools import partial
+import os
+from typing import Optional
+
+
+class MissingConfiguration(Exception):
+    def __init__(self, key: str, description: str):
+        self._key = key
+        self._description = description
+
+    def __str__(self):
+        return f"Missing configuration for {self._description}. Please set env var {self._key}."
+
+
+@dataclass
+class ApiConfig:
```

Comment:
Hmm, sorry if this is OT, but.

Will the `ApiConfig` work separate from all of the other `yapapi` stuff? E.g. can I do in my (otherwise non-yapapi) code do `from yapapi.config import ApiConfig` and use it?

If yes, that would be nice :)

Reply:
It was not intended but yes.

IDK if you would want to add yapapi dependency just to use a dataclass :sweat_smile:

Reply:
Well, maybe for development purposes at least?
Also, that might not be the only imported thing:
```
(gc) jbetley:~/golem-core-python {jb/yacat-v2} $ grep -rIh 'from yapapi' golem_core/
from yapapi.payload import Payload as YapapiPayload, vm
from yapapi import rest
from yapapi.engine import DEFAULT_DRIVER, DEFAULT_NETWORK, DEFAULT_SUBNET
from yapapi.props.builder import DemandBuilder
from yapapi import props
from yapapi.storage import Destination, Source
from yapapi.storage.gftp import GftpProvider
from yapapi.engine import NoPaymentAccountError
from yapapi.log import _YagnaDatetimeFormatter
from yapapi.engine import DEFAULT_NETWORK, DEFAULT_DRIVER, DEFAULT_SUBNET
from yapapi.props.base import constraint
from yapapi.props import inf
```
:)